zkvm/src/predicate.rs

```diff
@@ -52,7 +62,20 @@ impl Predicate {
     pub fn to_point(&self) -> CompressedRistretto {
         match self {
             Predicate::Opaque(p) => *p,
-            Predicate::Witness(w) => w.to_uncompressed_point().compress(),
+            Predicate::Key(k) => k.0,
+            Predicate::Or(l, r) => {
+                let l = l.to_point();
+                let r = r.to_point();
+                let f = Predicate::commit_or(l, r);
+                // l.decompress() + f * PedersenGens::default().B
+                unimplemented!()
```

Comment:
the need to decompress the point and desire to compute to_point w/o failing is what prompted creating `OpaqueBranch(RistrettoPoint)` originally.

Reply:
The failable conversion to point is bad because then `encode` must be failable and it poisons everything.

Reply:
Maybe try `Opaque(RistrettoPoint)` and fail on creation via helper method `Predicate::opaque(CompressedRistretto) -> Result<P, ...>`?

Reply:
But then it won't play nice with current design of PointOp that works with compressed points, so we'd need to redesign it too.

Reply:
Took another pass at still maintaining `RistrettoPoint` internally (but compressing in `to_point`, etc) !